File: programmers/ggyul/level1/최소직사각형/Solution2.java
// https://programmers.co.kr/learn/courses/30/lessons/86491
class Solution {
    public int solution(int[][] sizes) {
        // 긴 건 가로로, 짧은 건 세로로 맞춤
        int widthMax = 0;
        int heightMin = 0;
        for (int[] size : sizes) {
            widthMax = Math.max(Math.max(size[0], size[1]), widthMax);
            heightMin = Math.max(Math.min(size[0], size[1]), heightMin);
        }
        return widthMax * heightMin;
    }
}

File: programmers/ggyul/level2/단체사진_찍기/solution2.cpp
// https://programmers.co.kr/learn/courses/30/lessons/1835
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int answer = 0;
vector<string> condition;
string friends;

int getDistance(char from, char to) {
    int fromIdx = -1;
    int toIdx = -1;
    for (int i = 0; i < 8; i++) {
        if (friends[i] == from) {
            fromIdx = i;
        } else if (friends[i] == to) {
            toIdx = i;
        }
    }
    return abs(fromIdx - toIdx) - 1;
}

bool isPossible() {
    for (string d : condition) {
        char from = d[0];
        char to = d[2];
        char op = d[3];
        int dist = d[4] - '0';
        if (op == '=') {
            if (getDistance(from, to) != dist) return false;
        } else if (op == '<') {
            if (getDistance(from, to) >= dist) return false;
        } else if (op == '>') {
            if (getDistance(from, to) <= dist) return false;
        }
    }
    return true;
}

int solution(int n, vector<string> data) {
    answer = 0;
    condition = data;
    friends = "ACFJMNRT";
    
    do {
        if (isPossible())
            answer++;
    } while (next_permutation(friends.begin(), friends.end()));

    return answer;
}